mavros_extras/src/plugins/odom.cpp

```diff
@@ -66,90 +67,97 @@ class OdometryPlugin : public plugin::PluginBase {
 
 	void odom_cb(const nav_msgs::Odometry::ConstPtr &odom)
 	{
-		Eigen::Affine3d tr;
-		Eigen::Vector3d lin_vel_enu;
-		Eigen::Vector3d ang_vel_enu;
-
-		tf::poseMsgToEigen(odom->pose.pose, tr);
-		tf::vectorMsgToEigen(odom->twist.twist.linear, lin_vel_enu);
-		tf::vectorMsgToEigen(odom->twist.twist.angular, ang_vel_enu);
-
-		// get current ENU FCU orientation
-		Eigen::Quaterniond enu_orientation;
-		tf::quaternionMsgToEigen(m_uas->get_attitude_orientation(), enu_orientation);
-
-		// Build 9x9 covariance matrix to be transformed and sent
-		ftf::Covariance9d cov_full{};	// zero initialized
-		ftf::EigenMapCovariance9d cov_full_map(cov_full.data());
-
-		// 9x9 covariance matrix contruct
-		cov_full_map.block<6, 6>(0, 0) = ftf::EigenMapConstCovariance6d(odom->pose.covariance.data());
-		cov_full_map.block<3, 3>(6, 6) = Eigen::Matrix3d::Identity();
-
-		/* -*- vector transforms -*- */
-		// body frame rotations must be aware of current attitude of the vehicle
-		auto q_ned_current = ftf::transform_orientation_enu_ned(
-					ftf::transform_orientation_baselink_aircraft(enu_orientation));
-
-		// apply coordinate frame transforms
-		auto pos_ned = ftf::transform_frame_enu_ned(Eigen::Vector3d(tr.translation()));
-		auto lin_vel_ned = ftf::transform_frame_enu_ned(lin_vel_enu);
-		auto q_ned = ftf::transform_orientation_enu_ned(
-					ftf::transform_orientation_baselink_aircraft(Eigen::Quaterniond(tr.rotation())));
-
-		// angular velocity - WRT body frame
-		auto ang_vel_ned = ftf::transform_frame_enu_ned(
-					ftf::transform_frame_baselink_aircraft(
-						ftf::transform_frame_ned_aircraft(ang_vel_enu, q_ned_current)));
-
-		/* -*- covariance transforms -*- */
-		/**
-		 * @brief Pose+Accel 9-D Covariance matrix
-		 * WRT world frame (half upper right triangular)
-		 */
-		auto cov_full_tf = ftf::transform_frame_enu_ned(cov_full);
-
-		ftf::EigenMapConstCovariance9d cov_tf_map(cov_full_tf.data());
-		auto urt_view = Eigen::Matrix<double, 9, 9>(cov_full_map.triangularView<Eigen::Upper>());
-		ROS_DEBUG_STREAM_NAMED("odom", "Odometry: pose+accel covariance matrix: " << std::endl << cov_tf_map);
-		ROS_DEBUG_STREAM_NAMED("odom", "Odometry: Cov URT: " << std::endl << urt_view);
-
-		/**
-		 * @brief Velocity 6-D Covariance matrix
-		 * WRT world frame (half upper right triangular)
-		 */
-		auto cov_vel = ftf::transform_frame_enu_ned(
-					ftf::transform_frame_baselink_aircraft(
-						ftf::transform_frame_ned_aircraft(odom->twist.covariance, q_ned_current)));
-
-		ftf::EigenMapConstCovariance6d cov_vel_map(cov_vel.data());
-		ROS_DEBUG_STREAM_NAMED("odom", "Odometry: velocity covariance matrix: " << std::endl << cov_vel_map);
+		// lookup transforms
+		Eigen::Affine3d tfChild2NED;
+		Eigen::Affine3d tfParent2NED;
+		try {
+			tfChild2NED = tf2::transformToEigen(m_uas->tf2_buffer.lookupTransform(
```

Comment:
So basically this would be a function that looks up to the transform between the parent and the child where a static tf is applied (ENU<->NED). That static tf would have to be added to the launch files I suppose.